File: slavicmeme/week_12/BOJ_3273_두수의합.java
package Week12;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class BOJ_3273_두수의합 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int n = Integer.parseInt(br.readLine());
        
        int[] sequence = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++)
            sequence[i] = Integer.parseInt(st.nextToken());
        
        int x = Integer.parseInt(br.readLine());

        int left = 0;
        int right = n - 1;
        int count = 0;
        
        Arrays.sort(sequence);
        while (left < right) 
        {
            int sum = sequence[left] + sequence[right];
            if (sum == x) 
            {
                count++;
                left++;
                right--;
            } 
            else if (sum < x)
                left++;
            else
                right--;
        }
        
        System.out.println(count);
    }
}

File: youngjoo00/week_12/BOJ_3273_두 수의 합.swift
// BOJ_3273_두 수의 합.swift

let n = Int(readLine()!)!
let list = readLine()!.split(separator: " ").map { Int($0)! }.sorted()
let x = Int(readLine()!)!

// 1<i<j<n 의 수를 정확하게 서치하기 위해 오름차순 정렬 진행
// 두 수의 덧셈과 비교하기위해서 처음과 끝을 기준으로 좁혀가면서 서치해야함
var start = 0
var end = list.count - 1
var count = 0

// 더한 값이 작으면 start ++
// 더한 값이 크면 end ++
while start < end { // start 가 end 보다 작을때만 서치
    let sum = list[start] + list[end]
    if sum == x {
        start += 1
        count += 1
    } else if sum > x {
        end -= 1
    } else {
        start += 1
    }
}

print(count)